server/supernova/sc/sc_osc_handler.cpp

```diff
@@ -1985,21 +1985,30 @@ template <bool realtime> void handle_b_alloc(ReceivedMessage const& msg, endpoin
     osc::ReceivedMessageArgumentStream args = msg.ArgumentStream();
 
     osc::int32 bufferIndex, frames, channels;
+    float sampleRate;
 
     args >> bufferIndex >> frames;
 
-    if (!args.Eos())
+    if (!args.Eos()) {
         args >> channels;
-    else
+        if (!args.Eos()) {
+            args >> sampleRate;
+            if (sampleRate == 0.0)
+                sampleRate = sc_factory->world.mSampleRate;
+        } else
+            sampleRate = sc_factory->world.mSampleRate;
+    } else {
         channels = 1;
+        sampleRate = sc_factory->world.mSampleRate;
+    }
 
     completion_message message = extract_completion_message(args);
 
     cmd_dispatcher<realtime>::fire_system_callback([=, message = std::move(message)]() mutable {
         sc_ugen_factory::buffer_lock_t buffer_lock(sc_factory->buffer_guard(bufferIndex));
         try {
             sample* free_buf = sc_factory->get_nrt_mirror_buffer(bufferIndex);
-            sc_factory->allocate_buffer(bufferIndex, frames, channels);
+            sc_factory->allocate_buffer(bufferIndex, frames, channels, sampleRate);
```

Comment:
Make sure that `sc_plugin_interface::allocate_buffer` throws an error if the sample rate is <= 0, similar to how it handles frames/channels.